presentation-compiler/src/main/scala/org/alephium/ralph/lsp/pc/search/gotodef/soft/GoToDefIdentifier.scala

```diff
@@ -197,6 +202,196 @@ private object GoToDefIdentifier extends StrictImplicitLogging {
+            identifier = ident,
+            target = field,
+            sourceCode = sourceCode
+          )
+      }
+      .flatten
+
+  /**
+   * Search for `struct` definitions matching the constructor identifier used by the given
+   * struct-field assignment.
+   *
+   * {{{
+   *   struct >>MyStruct<< {
+   *     field: Type
+   *   }
+   *
+   *   @@MyStruct {
+   *     @@field: value // struct-field assignment
+   *   }
+   * }}}
+   *
+   * @param structField The struct-field assignment contained within the constructor.
+   * @param sourceCode  The block-part and its source code state where this search is executed.
+   * @param workspace   The workspace state where the source-code is located.
+   * @return Locations of all structs matching the assignment's constructor struct-identifier.
+   */
+  private def searchStruct(
+      structField: Node[SoftAST.StructFieldAssignment, SoftAST],
+      sourceCode: SourceLocation.CodeSoft,
+      workspace: WorkspaceState.IsSourceAware,
+      settings: GoToDefSetting
+    )(implicit searchCache: SearchCache,
+      logger: ClientLogger): Iterator[SourceLocation.NodeSoft[SoftAST.Struct]] =
+    structField
+      .walkParents
+      .take(3) // Search at most 3 parents because this field could be within the tail of a Group, which has `StructConstructor` as its grandparent (3rd position up).
+      .collectFirst {
+        case Node(constructor: SoftAST.StructConstructor, _) =>
+          searchStruct(
+            constructor = constructor,
+            sourceCode = sourceCode,
+            workspace = workspace,
+            settings = settings
+          )
+      }
+      .getOrElse(Iterator.empty)
+
+  /**
+   * Search for `struct` definitions matching the given constructor identifier.
+   *
+   * {{{
+   *   struct >>MyStruct<< {
+   *     field Type
+   *   }
+   *
+   *   My@@Struct { // struct constructor
+   *     field: value
+   *   }
+   * }}}
+   *
+   * @param constructor The constructor whose identifier is used to search the matching struct definitions.
+   * @param sourceCode  The block-part and its source code state where this search is executed.
+   * @param workspace   The workspace state where the source-code is located.
+   * @return Locations of all structs matching the assignment's constructor struct-identifier.
+   */
+  private def searchStruct(
+      constructor: SoftAST.StructConstructor,
+      sourceCode: SourceLocation.CodeSoft,
+      workspace: WorkspaceState.IsSourceAware,
+      settings: GoToDefSetting
+    )(implicit searchCache: SearchCache,
+      logger: ClientLogger): Iterator[SourceLocation.NodeSoft[SoftAST.Struct]] = {
+    // Find all definitions that match the given constructor's identifier.
+    // Note: These could result in definitions that may not be structs.
+    val definitions =
+      CodeProvider
+        .goToDefSoft
+        .search(
+          linePosition = constructor.identifier.index.middle.toLineRange(sourceCode.parsed.code).from,
```

Comment:
why do we need the `middle` here?

Reply:
#597 is the issue that `middle` is solving for `struct`'s case. Didn't realise that the issue had an easy fix.

I've [resolved](https://github.com/alephium/ralph-lsp/pull/598/commits/9266bcc39fc730d1d978f0cf270e6f77d7a3750c) that issue and [removed `middle`](https://github.com/alephium/ralph-lsp/pull/598/commits/c8e8f9478f2c478c8be0bfab3375e191bca920cc).